x/retry/retry.go

```diff
@@ -57,19 +62,30 @@ func WithPolicyProvider(provider PolicyProvider) MiddlewareOption {
 	})
 }
 
+// TallyScope sets a Tally scope that will be used to record retry metrics.
+func TallyScope(scope tally.Scope) MiddlewareOption {
```

Comment:
It was a conflict with the PolicyProvider interface which resulted in the "With" semantics, I can modify this to be similar.

The yarpc.Config just uses the name "Tally" I'll make the option "WithTally" for consistency.